ship/src/main/scala/ch/epfl/bluebrain/nexus/ship/projects/OriginalProjectContext.scala

```diff
@@ -0,0 +1,49 @@
+package ch.epfl.bluebrain.nexus.ship.projects
+
+import cats.effect.IO
+import ch.epfl.bluebrain.nexus.delta.sdk.projects.FetchContext
+import ch.epfl.bluebrain.nexus.delta.sdk.projects.model.{ApiMappings, ProjectContext}
+import ch.epfl.bluebrain.nexus.delta.sourcing.Transactors
+import ch.epfl.bluebrain.nexus.delta.sourcing.implicits._
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{Identity, ProjectRef}
+import doobie.implicits._
+import io.circe.generic.semiauto.deriveCodec
+import io.circe.syntax.EncoderOps
+import io.circe.{Codec, Json}
+
+/**
+  * Allows to keep track of the original project context which will change because of project renaming / patching
+  * configuration so as to be able to use them to expand iris in resource payload
+  */
+class OriginalProjectContext(xas: Transactors) extends FetchContext {
```

Comment:
You don't need to implement FetchContext for this?

Reply:
The only place it's needed is in FileSelf, so you could just split the interface. I think it's likely that a lot of places it would make sense to just pass `onRead` rather than all of these methods

Reply:
Most of the methods come from `FetchContext` and the import batch is a special use case, if we are to refactor it, it is outside the scope of this PR